solver/src/driver.rs

```diff
@@ -431,14 +433,17 @@ impl Driver {
             }
 
             tracing::info!("winning settlement: {:?}", settlement);
-            self.submit_settlement(settlement.clone()).await;
-            self.inflight_trades = settlement
-                .settlement
-                .settlement
-                .trades()
-                .iter()
-                .map(|t| t.order.order_meta_data.uid)
-                .collect::<HashSet<OrderUid>>();
+            if self.submit_settlement(settlement.clone()).await.is_ok() {
+                self.inflight_trades = settlement
+                    .settlement
+                    .settlement
+                    .trades()
+                    .iter()
+                    .map(|t| t.order.order_meta_data.uid)
+                    .collect::<HashSet<OrderUid>>();
+            } else {
+                self.inflight_trades = HashSet::new()
+            }
```

Comment:
Instead of creating a new hash set every time we can reuse the existing one which is more efficient.
```suggestion
            self.inflight_trades.clear();
            if self.submit_settlement(settlement.clone()).await.is_ok() {
                self.inflight_trades.extend(
                    settlement
                        .settlement
                        .settlement
                        .trades()
                        .iter()
                        .map(|t| t.order.order_meta_data.uid),
                );
            }
```

Reply:
Maybe the clear should even happen outside of the `if let Some(mut settlement) = rated_settlements` statement? This way we reset them even if there is no new solution found.